owasp-security-logging-logback/src/main/java/org/owasp/security/logging/mask/LUHNMaskingConverter.java

```diff
@@ -0,0 +1,59 @@
+package org.owasp.security.logging.mask;
+
+import ch.qos.logback.classic.pattern.ClassicConverter;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class LUHNMaskingConverter extends ClassicConverter {
+
+    private static final String CREDIT_CARD_PATTERN = "s*\\d{1,5}\\s*(?:-*\\s*,*\\d{1,5}\\s*){1,4}";
+    private static final Pattern pattern = Pattern.compile(CREDIT_CARD_PATTERN);
+
+    @Override
+    public String convert(ILoggingEvent iLoggingEvent) {
+        return mask(iLoggingEvent.getFormattedMessage());
+    }
+
+    private String mask(String formattedMessage) {
+        Matcher matcher = pattern.matcher(formattedMessage);
+        while (matcher.find()) {
+            String found = matcher.group();
+            if (luhnCheck(removeSpecials(found))) {
+                StringBuilder builder = new StringBuilder();
+                builder.append(removeSpecials(found.substring(0, 4)));
+                builder.append("**MASKED**");
+                builder.append(removeSpecials(found.substring(0, 4)));
```

Comment:
Also this looks like it is repeating the first 4 digits twice. 
```suggestion
                builder.append(found.substring(found.length() - 4));
```